france_connect/clients.py

```diff
@@ -0,0 +1,192 @@
+    FranceConnectHttpError,
+    FranceConnectInvalidTokenError,
+)
+
+
+class FranceConnect:
+    """
+    FranceConnect Client.
+
+    This class represents a client for interacting with the FranceConnect API.
+    It provides methods for authentication, token exchange, and user information retrieval.
+
+
+    Parameters
+    ----------
+    client_id (str): The client ID for the FranceConnect application.
+    client_secret (str): The client secret for the FranceConnect application.
+    scopes (List[Scopes]): The list of scopes to request during authentication.
+    post_login_callback_url (str): The URL to redirect to after successful login.
+    post_logout_callback_url (str): The URL to redirect to after logout.
+    fc_base_url (str): The base URL of the FranceConnect API.
+    fc_authorization_url (str, optional): The authorization URL of the FranceConnect API.
+    fc_token_url (str, optional): The token URL of the FranceConnect API.
+    fc_jwks_url (str, optional): The JWKS URL of the FranceConnect API.
+    fc_userinfo_url (str, optional): The userinfo URL of the FranceConnect API.
+    fc_logout_url (str, optional): The logout URL of the FranceConnect API.
+    """
+
+    def __init__(
+        self,
+        client_id: str,
+        client_secret: str,
+        scopes: List[Scopes],
+        post_login_callback_url: str,
+        post_logout_callback_url: str,
+        fc_base_url: str = "https://fcp-low.integ01.dev-franceconnect.fr",
+        fc_authorization_url: str = "/api/v2/authorize",
+        fc_token_url: str = "/api/v2/token",
+        fc_jwks_url: str = "/api/v2/jwks",
+        fc_userinfo_url: str = "/api/v2/userinfo",
+        fc_logout_url: str = "/api/v2/session/end",
+        timeout: int = 10,
+        verify_ssl: bool = True,
+        allow_redirects: bool = True,
+    ):
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.scopes = scopes
+        self.post_login_callback_url = post_login_callback_url
+        self.post_logout_callback_url = post_logout_callback_url
+        self.fc_base_url = fc_base_url
+        self.fc_authorization_url = fc_authorization_url
+        self.fc_token_url = fc_token_url
+        self.fc_jwks_url = fc_jwks_url
+        self.fc_userinfo_url = fc_userinfo_url
+        self.fc_logout_url = fc_logout_url
+        self.timeout = timeout
+        self.verify_ssl = verify_ssl
+        self.allow_redirects = allow_redirects
+
+    def generate_nonce(self) -> str:
+        """Generate a random nonce in hexadecimal string (64 characters).."""
+        return secrets.token_hex(48)
+
+    def generate_state(self) -> str:
+        """Generate a random state in hexadecimal string (64 characters)."""
+        return secrets.token_hex(48)
+
+    def get_authentication_url(
+        self, acr_values: List[ACRValues] = None, nonce: str = None, state: str = None
+    ) -> Tuple[str, str, str]:
+        """Return the authentication URL, nonce and state."""
```

Comment:
It is not obvious how to use this method and what it returns, a detailed docstring might help:

```
        """Return the authentication URL, nonce, and state.

        Parameters
        ----------
        acr_values: List[ACRValues], optional
            The list of ACR values used for the authentication process, defaults
            to `[ACRValues.EIDAS1]`.
        nonce: str, optional
            The `nonce` used for the authentication process, generated using
            `generate_nonce()` if not provided.
        state: str, optional
            The `state` used for the authentication process, generated using
            `generate_state()` if not provided.

        Returns
        -------
        Tuple[str, str, str]
            The authentication URL, `nonce` and `state`.
        """
```